lib/bundler/settings.rb

```diff
@@ -130,21 +134,41 @@ def with
     end
 
     # @local_config["BUNDLE_PATH"] should be prioritized over ENV["BUNDLE_PATH"]
+    # Always returns an absolute path to the bundle directory
+    # TODO: Document and refactor this method
     def path
       key  = key_for(:path)
       path = ENV[key] || @global_config[key]
+      set_path = ""
+      install_path = ""
 
       if path && !@local_config.key?(key)
         path = "#{path}/#{Bundler.ruby_scope}" if path != Bundler.rubygems.gem_dir
-        return path
+        set_path = path
       end
 
       if path = self[:path]
         path = "#{path}/#{Bundler.ruby_scope}" if path != Bundler.rubygems.gem_dir
-        File.expand_path(path)
+        set_path = path
+      else
+        set_path = File.join(@root, Bundler.ruby_scope)
+      end
+
+      if set_path == File.join(Bundler.settings.root, Bundler.ruby_scope)
+        # ?
```

Comment:
When I first wrote this code, line 157 seemed useful, but I forgot why I added it. I need to sit down, refactor, and comment this block of code later, because it's not very transparent right now.